libdnf5/base/transaction.cpp

```diff
@@ -448,6 +449,22 @@ void Transaction::Impl::set_transaction(
         TransactionPackage tspkg(pkg, TransactionPackage::Action::REASON_CHANGE, reason, group_id);
         packages.emplace_back(std::move(tspkg));
     }
+
+    // After all packages were added check rpm reason overrides
+    //TODO(amatej): Should these reason overrides be done via a REASON_CHANGE action? This would show up in the transaction table
+    if (!rpm_reason_overrides.empty()) {
+        for (auto & pkg : packages) {
+            const auto reason_override = rpm_reason_overrides.find(pkg.get_package().get_nevra());
+            if (reason_override != rpm_reason_overrides.end()) {
+                // For all other actions we don't want to lower the reason
```

Comment:
I am not sure about the condition, may be after finishing the review I will have a better picture.

Reply:
I will test some edge cases

Reply:
`// For all other actions we don't want to lower the reason`
May I ask you why?

What do you think  is more important  -  reason that is listed in transaction file or reason in system-state (internal DNF5 DB)?

Note:
Upgrade is an operation that normally does not change reason,  right? But when upgrading  package also obsoletes another package than the final reason is the strongest of both.

Reply:
I think the more important reason is the stronger one. Just like with the obsoletes.

Specifically there is a [test scenario](https://github.com/rpm-software-management/ci-dnf-stack/blob/main/dnf-behave-tests/dnf/transaction-sr/replay.feature#L1228-L1282) where the replayed transaction has an upgrade with `Dependency` but on host the package has `User` reason. I think it would be wrong to lower the reason to `Dependency`. The package is user-installed.

So I think we don't want to lower the reason because some user installed packages could get marked as dependencies.

Reply:
Thank you for the explanation. Looks good to me